src/sources/soundsourceproxy.cpp

```diff
@@ -485,64 +485,75 @@ mixxx::AudioSourcePointer SoundSourceProxy::openAudioSource(const mixxx::AudioSo
         } else {
-            kLogger.warning() << "Failed to open file"
-                              << getUrl().toString()
-                              << "with provider"
-                              << getSoundSourceProvider()->getName();
-            if (openResult == mixxx::SoundSource::OpenResult::Succeeded) {
-                m_pSoundSource->close(); // cleanup
+            kLogger.warning()
+                    << "Failed to open file"
+                    << getUrl().toString()
+                    << "with provider"
+                    << getSoundSourceProvider()->getName();
+            if (openMode == mixxx::SoundSource::OpenMode::Permissive) {
+                if (openResult == mixxx::SoundSource::OpenResult::Failed) {
+                    // Do NOT retry with the next SoundSource provider if the file
+                    // itself seems to be the cause when opening still fails during
+                    // the 2nd (= permissive) round.
+                    DEBUG_ASSERT(!m_pAudioSource);
+                    return m_pAudioSource;
+                } else {
+                    // Continue and give other providers the chance to open the file
+                    // in turn.
+                    DEBUG_ASSERT(openResult == mixxx::SoundSource::OpenResult::Aborted);
+                }
+            } else {
+                // Continue and give other providers the chance to open the file
+                // in turn, independent of why opening the file failed.
+                DEBUG_ASSERT(openMode == mixxx::SoundSource::OpenMode::Strict);
             }
-            // Do NOT retry with the next SoundSource provider if the file
-            // itself is the cause!
-            DEBUG_ASSERT(!m_pAudioSource);
         }
-        return m_pAudioSource; // either success or failure
+        kLogger.warning() << "Unable to open file"
+                          << getUrl().toString()
+                          << "with provider"
+                          << getSoundSourceProvider()->getName()
+                          << "using mode"
+                          << openMode;
+        // Continue with the next SoundSource provider
+        nextSoundSourceProvider();
+        if (!getSoundSourceProvider() &&
+                (openMode == mixxx::SoundSource::OpenMode::Strict)) {
+            // No provider was able to open the source in Strict mode.
+            // Retry to open the file in Permissive mode starting with
+            // the first provider...
+            m_soundSourceProviderRegistrationIndex = 0;
+            openMode = mixxx::SoundSource::OpenMode::Permissive;
+        }
+        initSoundSource();
+        // try again
     }
     // All available providers have returned OpenResult::Aborted when
     // getting here. m_pSoundSource might already be invalid/null!
-    kLogger.warning() << "Unable to decode file"
-                      << getUrl().toString();
+    kLogger.warning()
```

Comment:
Is this one necessary?
We have
* Opening File...
* Failed to open...
* Opening File...
* Failed to open...
* Unable to decode...

I would remove it.
Or rename it to "Give up"

A log improvement could be to count the attempts.

This helps to distinguish automatic attempts from user attempts.

Reply:
I will not rewrite this entire part. Log messages have been improved now. Finished here.